### ISSUE TITLE
Update travis-ci config example

### ISSUE THREAD

**Issue by vityafx** [rust-lang-nursery/rustfmt#2402]
The readme proposes this example config:

```yaml
language: rust
cache: cargo
before_script:
- export PATH="$PATH:$HOME/.cargo/bin"
- which rustfmt || cargo install rustfmt-nightly
script:
- cargo fmt -- --write-mode=diff
- cargo build
- cargo test
```

But it does not work for me:

1. `cache: cargo` makes travis think so that it is wrongly written and so the default config (ruby language) is used by travis.
2. The `cargo build`, I guess, is better to change to `cargo check`.
3. `cargo fmt` fails:

```
$ which rustfmt || cargo install rustfmt-nightly
/home/travis/.cargo/bin/rustfmt
0.07s$ cargo fmt -- --write-mode=diff
error: toolchain 'stable-x86_64-unknown-linux-gnu' does not have the binary `cargo-fmt`
```

[Here is the log](https://travis-ci.org/vityafx/toornament-rs/jobs/336053975) when I remove the `cache: cargo` line.

**Comment by davidalber** [comment]
I think this is a duplicate of #2285. The error is a little different in this case, but the workaround at the bottom that issue fixes this case for me.

I'm unclear if that workaround is the best option right now, although I don't have any reason to believe it's not. It does add a lot of time to a build. If it is the current best path, we should update [README.md](https://github.com/rust-lang-nursery/rustfmt/blob/master/README.md) to reflect this.

**Comment by vityafx** [comment]
Yes, tested that, even changed `cargo build` to `cargo check` but it still compiles too long.

**Comment by vityafx** [comment]
Travis build time increas from 2 minutes to 20 minutes. We should definitely do something.

**Comment by topecongiro** [comment]
I played around with this a bit, and replacing `cargo install` with `rustup component add` did mitigate the build time. A simple example would be something like:

```yml
language: rust
before_script:
  - rustup toolchain install nightly
  - rustup run nightly rustup component add rustfmt-preview
  - which rustfmt || cargo install --force rustfmt-nightly
script:
  - cargo +nightly fmt --all -- --write-mode=diff
  - cargo build
  - cargo test
```

This does not work if you need the specific version of rustfmt, though.

**Comment by topecongiro** [comment]
Closed via #2411.
